ServerCore/Helpers/MailHelper.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Mail;
+using Microsoft.Extensions.Configuration;
+
+/// <summary>
+/// Helper class for sending mail with Mailjet.
+/// </summary>
+public class MailHelper
+{
+    private string PublicSecret;
+    private string PrivateSecret;
+
+    public bool Enabled { get; private set; }
+    private bool IsDev;
+
+    private const string FromAddress = "msph@puzzlehunt.org";
+    private const string ToAddressForBCC = "msph-bcc@puzzlehunt.org";
+    
+    public static MailHelper Singleton { get; private set; }
+
+    public static void initialize(IConfiguration configuration, bool isDev)
+    {
+        Singleton = new MailHelper(configuration, isDev);
+    }
+
+    private MailHelper(IConfiguration configuration, bool isDev)
+	{
+        try
+        {
+            PublicSecret = configuration["Mailjet-Public"];
+            PrivateSecret = configuration["Mailjet-Private"];
+            Enabled = true;
+            IsDev = isDev;
+        }
+        catch
+        {
+            Enabled = false;
+        }
+    }
+
+    public void Send(string recipient, string subject, string body)
+    {
+        Send(new List<string> { recipient }, false, subject, body);
+    }
+
+    public void Send(IEnumerable<string> recipients, bool bcc, string subject, string body)
+    {
+        // See https://www.mailjet.com/docs/code/c/c-sharp for sample code from Mailjet.
+        if (!Enabled)
+        {
+            if (!IsDev)
+            {
+                throw new InvalidOperationException("Mail was not configured.");
+            }
+            WriteMailToConsole(recipients, subject, body);
+            return;
+        }
+        MailMessage msg = new MailMessage();
+
+        msg.From = new MailAddress(FromAddress);
+
+        if (bcc)
+        {
+            msg.To.Add(new MailAddress(ToAddressForBCC));
+            foreach (string recipient in recipients) {
+                try
+                {
+                    msg.Bcc.Add(new MailAddress(recipient));
+                }
+                catch (FormatException)
+                {
+                    ;  // ignore
+                }
+            }
+        }
+        else
+        {
+            foreach (string recipient in recipients) {
+                try
+                {
+                    msg.To.Add(new MailAddress(recipient));
```

Comment:
Should this either bcc players or send them separate messages in order to avoid reply-all problems?

Reply:
Sending many emails will make us hit our email quota that much faster. Bcc is the way to avoid reply-all. Since the only case where we'd want them to see each other's emails is if one team member was sending mail to the rest of the team (which isn't contemplated). So I'll make bcc the only choice unless there's a single email.

Reply:
I want a team application to *not* Bcc the team and the applicant. Can I do that?